packages/nameguard-js/src/securePrimaryName.ts

```diff
@@ -1,36 +1,54 @@
-import { ens_beautify, ens_normalize } from "@adraffy/ens-normalize";
+import { ens_beautify } from "@adraffy/ens-normalize";
 import { PublicClient } from "viem";
-import { SecurePrimaryNameResult } from "@namehash/nameguard";
+import { LocalSecurePrimaryNameResult } from "@namehash/nameguard";
 import { computeImpersonationStatus } from "./impersonation";
 import { lookupPrimaryName } from "./lookup";
+import { isEnsNormalized } from "./normalization";
 
+/**
+ * Analyzes the primary name associated with an address to determine if it is a potential impersonation attempt.
+ *
+ * @param address - The address to analyze.
+ * @param client - The viem client used to perform the primary name lookup.
+ * @returns A promise that resolves to a `SecurePrimaryNameResult` object containing the analysis results.
+ */
 export async function securePrimaryName(
   address: string,
   client: PublicClient,
-): Promise<SecurePrimaryNameResult> {
+): Promise<LocalSecurePrimaryNameResult> {
```

Comment:
The API result contains the full nameguard report. The local provider cannot compute the full report so we can: 1. return null; 2. have a different schema.
I chose the second option simply to avoid adding that null case.
@lightwalker-eth what do you think?

Reply:
@Carbon225 I see, thanks for sharing this background info :+1:

I continue to believe we shouldn't introduce a new special type for `LocalSecurePrimaryNameResult`.

Instead suggest we do something like the following:

1. Update the definition and docs for `SecurePrimaryNameResult` so that the `nameguard_result` field can also be `null` for the additional case that no NameGuard report was requested.
2. Update the API schema across Python / TypeScript for `getSecurePrimaryName` so that the API request includes an additional boolean param to optionally include a NameGuard report in the response in the case that the returned `primary_name_status` is NOT `no_primary_name` (primary name is found). This field can be optional in the API and can default to `true`. In TypeScript the `SecurePrimaryNameOptions` interface should be expanded to include an optional boolean value for manually controlling this setting.
3. Update the `NameGuardLocalProvider` interface such that it stops being an `interface` and instead becomes a `class` that `extends `NameGuardProvider`. The `NameGuardLocalProvider` class should then just override the behaviour of `NameGuardProvider` so that the following is true:
   a. The `securePrimaryName` function is resolved locally, rather than remotely via an API call.
   b. The local implementation of `securePrimaryName` throws an Error due to an unsupported request if the new API param to optionally include a NameGuardReport in the response is set to `true`.
4. Any ideas related to local providers should be removed from the NameGuard SDK and moved into the NameGuardJS package. If someone wants to use the NameGuard SDK they shouldn't be introducing any knowledge or dependency on NameGuardJS and its concepts / dependencies.

Let me know what you think :+1:

Reply:
Cool.
@lightwalker-eth so the default val of includeNameGuardReport should be true for API client and false for local provider? Or true for both, and you have to explicitly set it to false for the local provider?

Reply:
@lightwalker-eth Implemented.

Creating a nameguard js client works like this:

```typescript
import { createClient } from "@namehash/nameguard-js"; // instead of @namehash/nameguard
const publicClient = createPublicClient({
  chain: mainnet,
  transport: http(PROVIDER_URI_MAINNET),
});
const nameguard = createClient({ publicClient });
const data = await nameguard.getSecurePrimaryName(address);
```

You can pass `{ computeNameGuardReport: true/false }` to `nameguard.getSecurePrimaryName`.
**It defaults to true in nameguard and false in nameguard-js.**

Nameguard JS does the computation locally if and only if:

1. the network in request options is unspecified or equals the network given to the client constructor
2. computeNameGuardReport is false or unspecified

Otherwise, it calls the API.